src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_RefReadOnly.cs

```diff
@@ -2319,6 +2319,89 @@ public class Test
                 Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "in Test y").WithArguments("System.Runtime.CompilerServices.IsReadOnlyAttribute", ".ctor").WithLocation(11, 46));
         }
 
+        [Fact]
+        public void ExternMethodInParametersShouldHaveInAttribute_Simple()
+        {
+            var text = @"
+using System.Runtime.InteropServices;
+class T
+{
+    [DllImport(""Other.dll"")]
+    public static extern void M(in int p, int q);
+}";
+
+            CompileAndVerify(text, symbolValidator: module =>
+            {
+                var parameters = module.GlobalNamespace.GetTypeMember("T").GetMethod("M").GetParameters();
+                Assert.Equal(2, parameters.Length);
+
+                Assert.True(parameters[0].IsMetadataIn);
+                Assert.False(parameters[1].IsMetadataIn);
+            });
+        }
+
+        [Fact]
+        public void ExternMethodInParametersShouldHaveInAttribute_Complex()
+        {
+            var text = @"
+using System.Runtime.InteropServices;
+class T
+{
+    [DllImport(""Other.dll"")]
+    public static extern void M([ComAliasName(""p2"")]in int p, [ComAliasName(""q2"")]int q);
+}";
+
+            CompileAndVerify(text, symbolValidator: module =>
+            {
+                var parameters = module.GlobalNamespace.GetTypeMember("T").GetMethod("M").GetParameters();
+                Assert.Equal(2, parameters.Length);
+
+                Assert.True(parameters[0].IsMetadataIn);
+                Assert.False(parameters[1].IsMetadataIn);
+            });
+        }
+
+        [Fact]
+        public void ExternMethodInParametersShouldHaveInAttribute_UserDefinedIn()
+        {
+            var text = @"
+using System.Runtime.InteropServices;
+class T
+{
+    [DllImport(""Other.dll"")]
+    public static extern void M([In]in int p, int q);
+}";
+
+            CompileAndVerify(text, symbolValidator: module =>
+            {
+                var parameters = module.GlobalNamespace.GetTypeMember("T").GetMethod("M").GetParameters();
+                Assert.Equal(2, parameters.Length);
+
+                Assert.True(parameters[0].IsMetadataIn);
+                Assert.False(parameters[1].IsMetadataIn);
+            });
+        }
+        
+        [Fact]
+        public void NonExternMethodInParametersShouldNotHaveInAttribute()
+        {
+            var text = @"
+using System.Runtime.InteropServices;
+class T
+{
+    public static void M(in int p, int q) { }
+}";
+
+            CompileAndVerify(text, symbolValidator: module =>
+            {
+                var parameters = module.GlobalNamespace.GetTypeMember("T").GetMethod("M").GetParameters();
+                Assert.Equal(2, parameters.Length);
+
+                Assert.False(parameters[0].IsMetadataIn);
+                Assert.False(parameters[1].IsMetadataIn);
+            });
+        }
```

Comment:
Other tests to consider:

- When `in` is used in a delegate signature
- When a parameter has the signature `[Out] in int e`

Reply:
@jaredpar, the second was made illegal for 7.2

Reply:
https://github.com/dotnet/roslyn/pull/22645